### DISCUSSION TITLE
Render HTML in Edge Functions

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#31238] [Questions] [answered]
Has anyone been able to succesfully render HTML using Edge Functions? I keep getting raw HTML

`import { serve } from "https://deno.land/std@0.192.0/http/server.ts";
import { corsHeaders } from "../_shared/cors.ts";

serve((req: Request) => {
  // Handle CORS preflight requests
  if (req.method === "OPTIONS") {
    return new Response("ok", { headers: corsHeaders });
  }

  try {
    const html = `
      <!DOCTYPE html>
      <html>
        <head>
          <title>Installation Page</title>
          <style>
            body {
              font-family: Arial, sans-serif;
              max-width: 800px;
              margin: 0 auto;
              padding: 20px;
            }
          </style>
        </head>
        <body>
          <h1>Installation Page</h1>
          <p>Welcome to the installation page. The current time is: ${new Date().toISOString()}</p>
        </body>
      </html>
    `;

    return new Response(html, {
      headers: { ...corsHeaders, "Content-Type": "text/html" },
      status: 200,
    });

  } catch (error) {
    return new Response(`<html><body><h1>Error</h1><p>${error instanceof Error ? error.message : 'An unknown error occurred'}</p></body></html>`, {
      headers: { ...corsHeaders, "Content-Type": "text/html" },
      status: 500,
    });
  }
});
`

![2024-12-19 15 31 09](https://github.com/user-attachments/assets/33b88460-f75c-449c-aa13-e39c07625602)

**Comment by GaryAustin1** [comment, marked as answer]
From the edge function limits doc... 
![image](https://github.com/user-attachments/assets/e7be1292-9430-4862-9fb0-f94f45391e82)

**Reply by ghost** [reply]
Well that explains it. Think ill be using CF Workers. Thanks mate 👍
